### ISSUE TITLE
[Question] The minSdkVersion of the com.google.firebase.messaging.cpp library is 26. It fails when building.

### ISSUE THREAD

**Issue by pbs0512** [firebase/firebase-cpp-sdk#284]
### [READ] For Firebase Unity SDK question, please report to [Firebase Unity Sample](https://github.com/firebase/quickstart-unity/issues/new/choose)

Once you've read this section and determined that your issue is appropriate for this repository, please delete this section.

### [REQUIRED] Please fill in the following fields:

  * Pre-built SDK from the [website](https://firebase.google.com/download/cpp) or open-source from this repo: _____
  * Firebase C++ SDK version: **7.1.0**
  * Main Firebase Components in concern: **analytics, admob,  remoteConfig,  messaging,  dynamicLinks**
  * Other Firebase Components in use: **admob,  remoteConfig,  messaging,  dynamicLinks**
  * Platform you are using the C++ SDK on: **Windows 10**
  * Platform you are targeting: **Android**

### [REQUIRED] Please describe the question here:
The minSdkVersion of the com.google.firebase.messaging.cpp library is 26. It fails when building.

> Manifest merger failed : uses-sdk:minSdkVersion 19 cannot be smaller than version 26 declared in library [com.google.firebase.messaging.cpp:firebase_messaging_cpp:] d:\test\android\AndroidManifest.xml as the library might be using APIs not available in 19
        Suggestion: use a compatible library with a minSdk of at most 19,
                or increase this project's minSdk version to at least 26,
                or use tools:overrideLibrary="com.google.android.gms.messaging.cpp" to force usage (may lead to runtime failures)

**Comment by thienphuoc** [comment]
I'm facing this problem

```
uses-sdk:minSdkVersion 16 cannot be smaller than version 26 declared in library [com.google.firebase.messaging.cpp:firebase_messaging_cpp:] /Users/apple/RunningMan/proj.android/gradle/caches/transforms-2/files-2.1/eaa19b386e9a39b236ecd801e6db21c6/firebase_messaging_cpp/AndroidManifest.xml as the library might be using APIs not available in 16
	Suggestion: use a compatible library with a minSdk of at most 16,
		or increase this project's minSdk version to at least 26,
		or use tools:overrideLibrary="com.google.android.gms.messaging.cpp" to force usage (may lead to runtime failures)
```
Could you solve this issue?

**Comment by Kirstone** [comment]
I'm also facing this. I don't want minSdkVersion to be 26. I need to support and lower ones.

**Comment by chkuang-g** [comment]
Ah, I think it is caused by this line.
https://github.com/firebase/firebase-cpp-sdk/blob/0cf2f97d1a86b9cb0d3d01f85a954b83773438be/messaging/src/android/java/LibraryManifest.xml#L4

I am not entirely sure why we made such a jump before. I will need to consult with the team to get more information.

As workaround, if you are building from source, try to lower this number to the min version you need, build it and test if FCM still function properly.

I'm gonna mark this as a bug until we figure out the lowest `minSdkVersion` FCM actually needs.

**Comment by GaRaOne** [comment]
Is there anything new here? We are therefore stuck on version 6, because we are using prebuild libs.

**Comment by wannabelikeyoda** [comment]
I found the commit with increase minSdkVersion https://github.com/firebase/firebase-cpp-sdk/commit/ecc9be9ecaa09ecf5a78aea1690976d1d69eccc1
Still waiting for solution I'll try to downgrade my Firebase SDK to 6.11 https://github.com/firebase/firebase-cpp-sdk/releases/tag/v6.11.0

**Comment by theoriginalgri** [comment]
So it's 7 months after that commit and there's no explanation why this is needed?

When looking at the commit mentioned in the comment above, I can't figure out the reason. It introduced the usage of androidx and `androidx.core.app.JobIntentService` which clearly states in it's documentation there's a compatibility < 26 (Android O).

> Helper for processing work that has been enqueued for a job/service. When running on Android O or later, the work will be dispatched as a job via JobScheduler.enqueue. When running on older versions of the platform, it will use Context.startService.

Raising the `minSdkVersion` to 26 is a no-go for everyone who has to support older versions than that.

The changelog of the firebase c++ sdk also does not mention this change, instead there's only

> Updated the library to be compatible with Android O.

which is quite the opposite of what happened here. See https://firebase.google.com/support/release-notes/cpp-relnotes?hl=de#fcm_2

**Comment by mrpuzzler** [comment]
Any updates on this? Can't use latest Firebase.

**Comment by eduhub123** [comment]
HI, @chkuang-g  Any updates on this issue?

**Comment by alexames** [comment]
Hi all, sorry that this took so long. It should be fixed as of #421.
